src/api/app/models/comment.rb

```diff
@@ -98,5 +98,6 @@ def blank_or_destroy
   # FIXME: This is to work around https://github.com/rails/rails/pull/12450/files
   def destroy
     super
+    parent.destroy if parent && parent.user == User.find_nobody! && parent.children.length.zero?
```

Comment:
Maybe it is more clean to use ActiveModel callbacks:
http://guides.rubyonrails.org/active_record_callbacks.html#destroying-an-object